controllers/infrastructure.cluster.x-k8s.io/kopsmachinepool_controller.go

```diff
@@ -294,7 +288,7 @@ func GetASGByTag(kopsMachinePool *infrastructurev1alpha1.KopsMachinePool, awsCli
 	clusterFilterKey := "tag:KubernetesCluster"
 	instanceGroupKey := "tag:kops.k8s.io/instancegroup"
 
-	igName, err := getInstanceGroupNameFromKopsMachinePool(kopsMachinePool)
+	asgName, err := kops.GetAutoScalingGroupNameFromKopsMachinePool(*kopsMachinePool)
```

Comment:
This should be the kops instance group name, not the ASG name. Or we can update the input to describe the ASG below to use this asgName by replacing the filters

Reply:
Changed to get the ASG by name and filtering by cluster only, instead of only filtering by name and cluster.